compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -2830,6 +2834,18 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
             case _ => traverseChildren(t)
       }.traverse(tree)
 
+    private def yCheckValidExprs(terms: List[Term]): terms.type =
+      if yCheck then terms.foreach(yCheckValidExpr)
+      terms
+    private def yCheckValidExpr(termOpt: Option[Term]): termOpt.type =
+      if yCheck then termOpt.foreach(yCheckValidExpr)
+      termOpt
+    private def yCheckValidExpr(term: Term): term.type =
+      if yCheck then
+        assert(!term.tpe.widenDealias.isInstanceOf[dotc.core.Types.MethodicType],
+          "Reference to a method must be eta-expanded before it is used as an expression: " + term.show)
```

Comment:
For later: maybe provide a reflection API to perform eta-expansion?

Reply:
It seems such well-formedness checks are very useful in macros development.

Is it possible to centralize the check for the expanded tree? I guess for debugging, the current approach is more helpful, as it shows the stacktrace.

Reply:
> For later: maybe provide a reflection API to perform eta-expansion?

We already have `term.etaExpand`

Reply:
> Is it possible to centralize the check for the expanded tree?

Centralize how? Like in normal `yCheck`?

Reply:
> Centralize how? Like in normal `yCheck`?

Yes, something like a `TreeMap` or `ReTyper`. But as I mentioned above, this might not be a good idea, as it cannot show the stacktrace how an ill-formed tree is created.

Reply:
That did not work in this case because the crash happened before we finished the transformation in the `Inlining` phase.